spec.html

```diff
@@ -1,19 +1,86 @@
-contributors: Ben Coe, Daniel Ehrenberg, Robert Kieffer, Roman Shtylman, Christoph Tavan
+<link rel="icon" href="favicon.ico">
+<link href="ecmarkup.css" rel="stylesheet">
+<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/8.4/styles/github.min.css">
+<script src="ecmarkup.js"></script>
+<style>
+ins .hljs.hljs {
+  background-color: inherit;
+}
+</style>
+<script>
+  if (location.hostname.endsWith('.github.io') && location.protocol !== 'https:') {
+    location.protocol = 'https:';
+  }
+</script>
+<pre class=metadata>
+  title: UUID
+  status: proposal
+  stage: 2
+  location: https://github.com/tc39/proposal-uuid
+  copyright: false
+  contributors: Benjamin Coe, Robert Kieffer, Christoph Tavan
 </pre>
-<script src="ecmarkup.js" defer></script>
-<link rel="stylesheet" href="ecmarkup.css">
 
-<emu-intro id="introduction">
-  <h1>Introduction</h1>
-    <p>The standard module _@std:uuid_ exposes an API for generating character
-    encoded Universally Unique IDentifiers (UUID), based on the IETF
-    <a href="https://tools.ietf.org/html/rfc4122">rfc4122 proposal</a></p>
-</emu-intro>
+<emu-clause id="sec-function-properties-of-the-global-object">
+  <h1><span class="secnum">18.2</span>Function Properties of the Global Object</h1>
+  <emu-clause id="sec-eval-x">
+    <h1><span class="secnum">18.2.1</span>eval ( <var>x</var> )</h1>
+    <p>The <code>eval</code> function is the  <dfn>%eval%</dfn> intrinsic object. When the <code>eval</code> function is called with one argument <var>x</var>, the following steps are taken:</p>
+    <emu-alg><ol><li><emu-xref href="#assert" id="_ref_4794"><a href="#assert">Assert</a></emu-xref>: The <emu-xref href="#execution-context-stack" id="_ref_4795"><a href="#execution-context-stack">execution context stack</a></emu-xref> has at least two elements.</li><li>Let <var>callerContext</var> be the second to top element of the <emu-xref href="#execution-context-stack" id="_ref_4796"><a href="#execution-context-stack">execution context stack</a></emu-xref>.</li><li>Let <var>callerRealm</var> be <var>callerContext</var>'s <emu-xref href="#realm" id="_ref_4797"><a href="#realm">Realm</a></emu-xref>.</li><li>Let <var>calleeRealm</var> be <emu-xref href="#current-realm" id="_ref_4798"><a href="#current-realm">the current Realm Record</a></emu-xref>.</li><li>Perform ?&nbsp;<emu-xref aoid="HostEnsureCanCompileStrings" id="_ref_4799"><a href="#sec-hostensurecancompilestrings">HostEnsureCanCompileStrings</a></emu-xref>(<var>callerRealm</var>, <var>calleeRealm</var>).</li><li>Return ?&nbsp;<emu-xref aoid="PerformEval" id="_ref_4800"><a href="#sec-performeval">PerformEval</a></emu-xref>(<var>x</var>, <var>calleeRealm</var>, <emu-val>false</emu-val>, <emu-val>false</emu-val>).
+    </li></ol></emu-alg>
+  </emu-clause>
+
+  <ins class="block">
+    <emu-clause id="sec-eval-x">
+      <h1><span class="secnum">18.2.7</span>randomUUID()</h1>
+      <p>Produces a 128 bit RFC-4122 UUID string.</p>
```

Comment:
My personal feeling is that we should _not_ write out any details of the version 4 UUID algorithm here but instead reference the RFC as much as possible.

The only thing that we should explicitly specify, is that implementations _MUST_ use `Math.getRandomValues()` as their source of randomness, otherwise we defeat the purpose of only adding _one_ additional source of randomness.

This is only my personal opinion and I'm really not experienced at all with writing spec text, so I'm curious to hear what the pros will say.

Reply:
👍 I agree with not going too deep into describing the algorithm, but I do think we need to describe inputs, outputs, and error behavior for both methods?

Reply:
i think that we definitely need to describe the interface behavior, at the least.